### PR TITLE
MAGN-9631 Large value range expression crashes on committing the CBN.

### PR DESCRIPTION
**Pull request by ke-yu**
### Purpose

To fix defect [MAGN-9631 Large value range expression crashes on committing the CBN](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-9631).

Internally the engine has been able to catch the out-of-memory exception that potentially thrown out during memory allocation (`Heap.AllocateXXX()` functions and memory expansion). But there are still couple of places where out-of-memory exception could still happen.

For this defect, one place is in `RuntimeMirror` which calls `ExecutionMirror` to unpack the internal value to C# object (in `ExecutionMirror.GetValue()`). Memory allocation happens if the internal value is an array.
```
StackValue svData = deprecateThisMirror.GetValue(variableName, blockDeclaration).DsasmValue;
```
 But as only `StackValue` is needed, this PR removes `ExecutionMirror.GetValue()` and replaces it with `ExecutionMirror.GetSymbolValue()` which only returns `StackValue`.

The other place is `MirrorData.GetElements()`. Internally it allocates a list to hold all values. This PR changes it to `IEnumerable<>`. 

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.

### Reviewers

@sharadkjaiswal 

### FYIs
@jnealb @riteshchandawar @monikaprabhu

### CODE REVIEW

File: src/Engine/ProtoCore/Reflection/Mirror.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ public RuntimeMirror(string varname, int blockDecl, ProtoCore.RuntimeCore runtim
 
                 variableName = varname;
                 blockDeclaration = blockDecl;
-                StackValue svData = deprecateThisMirror.GetValue(variableName, blockDeclaration).DsasmValue;
```
Comment (ke-yu) [inline src/Engine/ProtoCore/Reflection/Mirror.cs]: If the value for this variable is an array, `GetValue()` will convert it to C# list. But we are only interested in its `StackValue`

File: src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs [hunk 2]
```diff
@@ -357,7 +357,7 @@ private static void AddToLabelMap(MirrorData data, List<string> map, string tag)
             {
                 AddToLabelMap(data.Data as IEnumerable, map, tag);
             }
-            else
+            else if (data.Data is IGraphicItem)
```
Comment (ke-yu) [inline src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs]: Is this condition too strong?

File: src/Engine/ProtoCore/DSASM/Mirror/ExecutionMirror.cs [hunk 3]
```diff
@@ -723,23 +710,21 @@ public Obj GetWatchValue()
             return retVal;
         }
         
-        public Obj GetValue(string name, int block = 0, int classcope = Constants.kGlobalScope)
+        /// <summary>
+        /// Return the stack value of specified symbol.
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/DSASM/Mirror/ExecutionMirror.cs]: It seems that this method just looks for Symbol in global scope?? If so can we update the method name as well as documentation here to reflect that?

File: src/Engine/ProtoCore/DSASM/Mirror/ExecutionMirror.cs [hunk 4]
```diff
@@ -1389,68 +1306,6 @@ public static StackValue Repack(Obj obj, ProtoCore.DSASM.Heap heap)
             return obj.DsasmValue;
         }
 
-        public bool CompareArrays(DsasmArray dsArray, List<Object> expected, System.Type type)
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/DSASM/Mirror/ExecutionMirror.cs]: Can we get rid of Obj and DsasmArray types??

File: src/Engine/ProtoCore/Reflection/Mirror.cs [hunk 5]
```diff
@@ -66,7 +66,7 @@ public RuntimeMirror(string varname, int blockDecl, ProtoCore.RuntimeCore runtim
...
+                StackValue svData = deprecateThisMirror.GetSymbolValue(variableName);
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Reflection/Mirror.cs]: How is going to handle variables declared in nested blocks, especially using Imperative block?

File: src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs (hunk 2 above)
Comment (sharadkjaiswal) [inline src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs]: This should be fine, but ideally this change should be part of a different PR.

File: src/Engine/ProtoCore/Reflection/Mirror.cs (hunk 5 above)
Comment (ke-yu) [inline src/Engine/ProtoCore/Reflection/Mirror.cs]: In imperative block, all variables that are out of scope will be garbage collected. The only place that I can think of is debugging where the debugger is in the inner scope.

File: src/Engine/ProtoCore/DSASM/Mirror/ExecutionMirror.cs (hunk 3 above)
Comment (ke-yu) [inline src/Engine/ProtoCore/DSASM/Mirror/ExecutionMirror.cs]: Noted. Will update.

File: src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs (hunk 2 above)
Comment (ke-yu) [inline src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs]: This place causes out-of-memory because it adds all elements to `map`.
